### ISSUE TITLE
Can we use github plugin to backup our vault along with livesync plugin?

### ISSUE THREAD

**Issue by Gourav1910** [vrtmrz/obsidian-livesync#232]
As you mentioned in the documentation that don't use two or more syncronisation method when we use livesync but you also mentioned that this is not a backup solution. So can we use github backup solution along with this plugin?

**Comment by vrtmrz** [comment]
Thank you for asking me!
Yes, our vault can be backed up via Git or other synchronising solutions. However, we should not activate more than one synchronisation solution at the same time.
Self-hosted LiveSync does not use any special files for synchronisation. But other solutions may use some. Please make sure that the files will be ignored when you use `Hidden file synchronisation`. (The default configuration contains `.git`).

Just a quick supplement, If the other synchronisation has not been performed continuously or in real-time, mostly harmless now. (i.e., Git, FolderSync on Android). However, OneDrive, iCloud and like them still could cause a nightmare.

**Comment by Gourav1910** [comment]
> Thank you for asking me! Yes, our vault can be backed up via Git or other synchronising solutions. However, we should not activate more than one synchronisation solution at the same time. Self-hosted LiveSync does not use any special files for synchronisation. But other solutions may use some. Please make sure that the files will be ignored when you use `Hidden file synchronisation`. (The default configuration contains `.git`).
> 
> Just a quick supplement, If the other synchronisation has not been performed continuously or in real-time, mostly harmless now. (i.e., Git, FolderSync on Android). However, OneDrive, iCloud and like them still could cause a nightmare.

I am sorry but I am still confused. Let me rephrase my question again. So there is also a plugin of GitHub which backup our vault to GitHub. And then there is livesync plugin that you have created. So as I understood livesync plugin would be useful for sync file across devices such windows to android. But it don't backup file as GitHub plugin do on GitHub. So can I use these two plugin together? So that GitHub plugin will backup my data on GitHub and your plugin (livesync) will sync my data(vault) across devices?

**Comment by vrtmrz** [comment]
Sorry for trying to explain in detail and making it more difficult to do so...

We can use both Self-hosted LiveSync and Obsidian-git.

To be safer, following rules help us more safe.
- Do not start Self-hosted LiveSync's synchronisation while Obsidian-git is working. 
- And also, Do not start Obsidian-git synchronisation while Self-hosted LiveSync is processing files.

If you want to enable `Hidden file sync` in Self-hosted LiveSync, make sure that `/.git/` is in `Skip Patterns` which is on the `Sync Settings` pane.

Please avoid performing two synchronisations at the exact same time.

**Comment by Gourav1910** [comment]
Ok so I think you are trying to say that we can use livesync plugin all the time but if we wanna backup our data so we should turn of the livesync plugin for a moment and turn on the obsidian git plugin and update the backup and then again turn it off and turn on the livesync plugin. So we can do this once a day so in that way we can also take our backup daily and also we can enjoy the livesync plugin. Am I correct in my understanding?

**Comment by vrtmrz** [comment]
I apologise for being so much late!
Yes, you are correct surely!

**Comment by danielo515** [comment]
I think the gist is to use the git or github plugins only for backup, and never use it for sync and leave the sync part to this plugin

**Comment by Gourav1910** [comment]
> I think the gist is to use the git or github plugins only for backup, and never use it for sync and leave the sync part to this plugin

You mean we can use both at the same time?

**Comment by danielo515** [comment]
> > I think the gist is to use the git or github plugins only for backup, and never use it for sync and leave the sync part to this plugin
> 
> You mean we can use both at the same time?

It all depends on what you mean "at the same time".
If you mean just have both plugins installed, yes, you can use them at the same time.
If you mean: Have git automatically sync (pull and push) and this plugin automatically sync, then no, you can't.
I have git just for backups, I have it set to NEVER do anything automatically and I manually backup it once a day or so. Then, in this scenario, yes, you can use both

**Comment by Gourav1910** [comment]
> > > I think the gist is to use the git or github plugins only for backup, and never use it for sync and leave the sync part to this plugin
> > 
> > 
> > You mean we can use both at the same time?
> 
> It all depends on what you mean "at the same time". If you mean just have both plugins installed, yes, you can use them at the same time. If you mean: Have git automatically sync (pull and push) and this plugin automatically sync, then no, you can't. I have git just for backups, I have it set to NEVER do anything automatically and I manually backup it once a day or so. Then, in this scenario, yes, you can use both

Ok thankuu now i understood clearly. I just have to turn off the automatic backup in git plugin and do it manually once a day. Can you show the screenshot of how you setup livesync plugin because its setting looks so complicated. I am not able to setup it yet.

**Comment by Gourav1910** [comment]
> > > I think the gist is to use the git or github plugins only for backup, and never use it for sync and leave the sync part to this plugin
> > 
> > 
> > You mean we can use both at the same time?
> 
> It all depends on what you mean "at the same time". If you mean just have both plugins installed, yes, you can use them at the same time. If you mean: Have git automatically sync (pull and push) and this plugin automatically sync, then no, you can't. I have git just for backups, I have it set to NEVER do anything automatically and I manually backup it once a day or so. Then, in this scenario, yes, you can use both

![image](https://github.com/vrtmrz/obsidian-livesync/assets/101202140/a0313e3a-5fab-4408-bd88-61cd2b66012b)
Ok I set it to 0 which means it disabled automatic backup. Is it correct?

**Comment by Gourav1910** [comment]
> I apologise for being so much late! Yes, you are correct surely!

![image](https://github.com/vrtmrz/obsidian-livesync/assets/101202140/9827e0a9-329d-43a7-90b7-03d617f64069)
Hi you are saying in documentation that "Enter the information for the database we have set up" and then you are hiding the password. Then how can I supposed to know what should I type there? Please clarify it.

**Comment by danielo515** [comment]
> And for manual backup do I need to first push then stage all and then commit. Right?

You can just do git backup and it will do it all
<img width="751" alt="image" src="https://github.com/vrtmrz/obsidian-livesync/assets/2270425/622f9071-6fad-482f-9e75-c8541a5c8a09">

**Comment by Gourav1910** [comment]
> > And for manual backup do I need to first push then stage all and then commit. Right?
> 
> You can just do git backup and it will do it all <img alt="image" width="751" src="https://user-images.githubusercontent.com/2270425/257816542-622f9071-6fad-482f-9e75-c8541a5c8a09.png">

Ok that is simple. Thankuu 😊

**Comment by Gourav1910** [comment]
> > Hi you are saying in documentation that "Enter the information for the database we have set up" and then you are hiding the password. Then how can I supposed to know what should I type there? Please clarify it.
> 
> There is no way to give specific instructions there. It all depends on how you configure you database. If you have an admin user, just put the password of that admin user. Admin rights are preferred because that way the plugin will be able to create the database and fix some potential issues

Now it feels like I will never able to use this plugin because till now I am completely blank about what should I do. There must be a demo guide for non technical user who know nothing about technology and want to setup. 😭😭😭

**Comment by Gourav1910** [comment]
> > Hi you are saying in documentation that "Enter the information for the database we have set up" and then you are hiding the password. Then how can I supposed to know what should I type there? Please clarify it.
> 
> There is no way to give specific instructions there. It all depends on how you configure you database. If you have an admin user, just put the password of that admin user. Admin rights are preferred because that way the plugin will be able to create the database and fix some potential issues

Can you give an example about what information I put here. And how I am going to connect it with mobile. Do it need to put the same information there as well?

**Comment by danielo515** [comment]
> Can you give an example about what information I put here. And how I am going to connect it with mobile. Do it need to put the same information there as well?

First, what are you using? your own couchdb server or cloudant?

**Comment by Gourav1910** [comment]
> > Can you give an example about what information I put here. And how I am going to connect it with mobile. Do it need to put the same information there as well?
> 
> First, what are you using? your own couchdb server or cloudant?

I don't know what does it mean. I use syncthing to sync my laptop and mobile. And use github plugin for backup.

**Comment by danielo515** [comment]
> I don't know what does it mean. I use syncthing to sync my laptop and mobile. And use github plugin for backup.

This plugin has nothing to do wyth synthing. It uses a database, known as couchdb to sync in real time between devices

**Comment by Gourav1910** [comment]
> > I don't know what does it mean. I use syncthing to sync my laptop and mobile. And use github plugin for backup.
> 
> This plugin has nothing to do wyth synthing. It uses a database, known as couchdb to sync in real time between devices

In simple language what i need to do to use this plugin if i know nothing about it? Can you send me a screen recording of setting it up. It would be helpful.

**Comment by shayded-exe** [comment]
> In simple language what i need to do to use this plugin if i know nothing about it? Can you send me a screen recording of setting it up. It would be helpful.

@Gourav1910 You need to be willing to do a bit of research. Have you even tried googling "what is couchdb?" or "how do i setup a couchdb server?"
If you really don't know what any of this means, you're going to need a LOT more handholding than you should ask from someone who does this for free. Either put in the basic effort yourself, or offer to pay the person for their time to hold your hand.

>  There must be a demo guide for non technical user who know nothing about technology and want to setup

If you want an experience like that I recommend you use a paid service with actual customer support. They get paid to hold your hand.
If you really have that little knowledge about technology, you are very likely to shoot yourself in the foot at some point trying to manage your own data.

**Comment by Gourav1910** [comment]
> > In simple language what i need to do to use this plugin if i know nothing about it? Can you send me a screen recording of setting it up. It would be helpful.
> 
> @Gourav1910 You need to be willing to do a bit of research. Have you even tried googling "what is couchdb?" or "how do i setup a couchdb server?" If you really don't know what any of this means, you're going to need a LOT more handholding than you should ask from someone who does this for free. Either put in the basic effort yourself, or offer to pay the person for their time to hold your hand.
> 
> > There must be a demo guide for non technical user who know nothing about technology and want to setup
> 
> If you want an experience like that I recommend you use a paid service with actual customer support. They get paid to hold your hand. If you really have that little knowledge about technology, you are very likely to shoot yourself in the foot at some point trying to manage your own data.

Are you trying to roast me or something? I was asking from a normal obsidian user perspective. Even dataview and templater plugins are also heavily technical but still they made it so simple through their documentation that even I can use it very easily. If you are making a plugin then you should put necessary documentation for everyone or else metion in clear way that for whom this plugin is for so that everyone don't waste their time trying to figure out things. And that's the purpose of the plugin itself that is to make things so much easier for the audience so that they don't have to step into technicalities and only focus on usecase. Otherwise why are even making a plugin itself. Just write a note somewhere that if you want additional features then get educated do Google research learn various coding languages do trial and error and all of that. Then why the whole market exist in the world. Why just we start saying to the people that learn and do everything on your own. So if I have no technical knowledge then I have full rights to ask anything whatever I want. Whether you help or not it's up to you.
